File: mtfe-cal.py
import datetime

def calculate_total_amount():
    # Get input values
    main_amount = int(input("Main Amount: "))
    start_date_str = input("Start Date (YYYY-MM-DD): ")
    end_date_str = input("End Date (YYYY-MM-DD): ")

    # Convert date strings to datetime objects
    start_date = datetime.datetime.strptime(start_date_str, "%Y-%m-%d").date()
    end_date = datetime.datetime.strptime(end_date_str, "%Y-%m-%d").date()

    # Calculate date range in days
    date_range = (end_date - start_date).days
    cal_range = (date_range * 2) // 3

    # Check main amount level
    if 26 <= main_amount <= 60:
        profit_per_day = 0.7
    elif 61 <= main_amount <= 200:
        profit_per_day = 1.5
    elif 201 <= main_amount <= 500:
        profit_per_day = 5.5
    elif 501 <= main_amount <= 900:
        profit_per_day = 12
    elif 901 <= main_amount <= 1500:
        profit_per_day = 20
    elif 1501 <= main_amount <= 2000:
        profit_per_day = 35
    elif 2001 <= main_amount <= 3500:
        profit_per_day = 43
    elif 3501 <= main_amount <= 4000:
        profit_per_day = 76
    elif 4001 <= main_amount <= 5000:
        profit_per_day = 87
    elif 5001 <= main_amount <= 8000:
        profit_per_day = 127
    elif 8001 <= main_amount <= 10000:
        profit_per_day = 174
    elif 10001 <= main_amount <= 12000:
        profit_per_day = 292
    elif 12001 <= main_amount <= 20000:
        profit_per_day = 306
    elif 20001 <= main_amount <= 30000:
        profit_per_day = 600
    elif 30001 <= main_amount <= 50000:
        profit_per_day = 900
    elif main_amount > 50001:
        profit_per_day = 1500
    else:
        print("Invalid main amount")
        return

    # Calculate total amount
    total_amount = main_amount
    allocated_date_range = cal_range

    while allocated_date_range > 0:
        total_amount += profit_per_day
        allocated_date_range -= 1

    # Display total amount
    print(f"Total Amount: {total_amount}\nSelected date count: {date_range}")

calculate_total_amount()

File: mtfe-cal.php
<?php

function calculateTotalAmount() {
    // Get input values
    $mainAmount = (int) readline("Main Amount: ");
    $startDate = readline("Start Date (YYYY-MM-DD): ");
    $endDate = readline("End Date (YYYY-MM-DD): ");

    // Convert date strings to DateTime objects
    $startDateTime = new DateTime($startDate);
    $endDateTime = new DateTime($endDate);

    // Calculate date range in days
    $dateInterval = $endDateTime->diff($startDateTime);
    $dateRange = $dateInterval->days;
    $calRange = ($dateRange * 2) / 3;

    // Check main amount level
    $profitPerDay;
    if ($mainAmount >= 26 && $mainAmount <= 60) {
        $profitPerDay = 0.7;
    } else if ($mainAmount > 61 && $mainAmount <= 200) {
        $profitPerDay = 1.5;
    } else if ($mainAmount > 201 && $mainAmount <= 500) {
        $profitPerDay = 5.5;
    } else if ($mainAmount > 501 && $mainAmount <= 900) {
        $profitPerDay = 12;
    } else if ($mainAmount > 901 && $mainAmount <= 1500) {
        $profitPerDay = 20;
    } else if ($mainAmount > 1501 && $mainAmount <= 2000) {
        $profitPerDay = 35;
    } else if ($mainAmount > 2001 && $mainAmount <= 3500) {
        $profitPerDay = 43;
    } else if ($mainAmount > 3501 && $mainAmount <= 4000) {
        $profitPerDay = 76;
    } else if ($mainAmount > 4001 && $mainAmount <= 5000) {
        $profitPerDay = 87;
    } else if ($mainAmount > 5001 && $mainAmount <= 8000) {
        $profitPerDay = 127;
    } else if ($mainAmount > 8001 && $mainAmount <= 10000) {
        $profitPerDay = 174;
    } else if ($mainAmount > 10001 && $mainAmount <= 12000) {
        $profitPerDay = 292;
    } else if ($mainAmount > 12001 && $mainAmount <= 20000) {
        $profitPerDay = 306;
    } else if ($mainAmount > 20001 && $mainAmount <= 30000) {
        $profitPerDay = 600;
    } else if ($mainAmount > 30001 && $mainAmount <= 50000) {
        $profitPerDay = 900;
    } else if ($mainAmount > 50001) {
        $profitPerDay = 1500;
    } else {
        echo "Invalid main amount\n";
        return;
    }

    // Calculate total amount
    $totalAmount = $mainAmount;
    $allocatedDateRange = $calRange;

    while ($allocatedDateRange > 0) {
        $totalAmount += $profitPerDay;
        $allocatedDateRange--;
    }

    // Display total amount
    echo "Total Amount: " . $totalAmount . "\n";
    echo "Selected date count: " . $dateRange . "\n";
}

calculateTotalAmount();
